### PR TITLE
common: Replace StringDataCompare with slices.Contains and cleanup string funcs

### PR DESCRIPTION
**Pull request by thrasher-**
# PR Description

Replace StringDataCompare with slices.Contains + a bit of a cleanup.  I've made StringSliceDifference generic since I thought it might be useful moving forward, though since we don't currently use it anywhere now I'd be happy to rm -rf it 😈 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/coinbasepro/coinbasepro.go [hunk 1]
```diff
@@ -224,9 +225,8 @@ func (c *CoinbasePro) GetHistoricRates(ctx context.Context, currencyPair, start,
 		values.Set("end", "")
 	}
 
-	allowedGranularities := [6]int64{60, 300, 900, 3600, 21600, 86400}
-	validGran, _ := common.InArray(granularity, allowedGranularities)
-	if !validGran {
+	allowedGranularities := []int64{60, 300, 900, 3600, 21600, 86400}
```
Comment (shazbert) [inline exchanges/coinbasepro/coinbasepro.go:228]: NOTE:  Just had a quick play and which I thought was interesting, we can reduce the two parameter instance of a lot of these. Out of scope though. 

```go
// Container is a generic type for slices
type Container[T comparable] []T

func (c Container[T]) Contains(in T) bool {
	return slices.Contains(c, in)
}

func TestXxx(t *testing.T) {
	allowedGranularities := Container[int64]{60, 300, 900, 3600, 21600, 86400}
	assert.True(t, allowedGranularities.Contains(300))
}
```

**Review by shazbert — CHANGES_REQUESTED** [review]
Good stuff just a minor thing, commentary and suggestion.

File: docs/ADD_NEW_EXCHANGE.md [hunk 2]
```diff
@@ -267,9 +267,9 @@ func TestGetEnabledExchanges(t *testing.T) {
 		t.Errorf("received: '%v' but expected '%v'", len(exchanges), defaultEnabledExchanges)
 	}
 
-	if !common.StringDataCompare(exchanges, "Bitfinex") {
+	if !slices.Contains(exchanges, "Bitfinex") {
```
[the remark this answers is not in the document]
Comment (thrasher-) [reply docs/ADD_NEW_EXCHANGE.md]: I removed these since they aren't relevant anymore, can do a full updated pass in #1494 once Bitget PR is opened

**Review by shazbert — APPROVED** [review]
🔥 Thanks. chACK.

File: common/common.go [hunk 3]
```diff
@@ -136,66 +135,47 @@ func NewHTTPClientWithTimeout(t time.Duration) *http.Client {
...
-					found = true
-					break
-				}
-			}
-			if !found {
-				diff = append(diff, s1)
-			}
+// SliceDifference returns the elements that are in slice1 or slice2 but not in both
+func SliceDifference[T comparable](slice1, slice2 []T) []T {
+	diff := make([]T, 0, len(slice1)+len(slice2))
```
Comment (shazbert) [inline common/common.go:140]: I gave you a silly billy suggestion. It gets gets weird when there are duplications, and its modifying the incoming slice. 
Just the prealloc is fine. 🙏 
```diff
diff --git a/common/common.go b/common/common.go
index d0799b199..57e0a92d2 100644
--- a/common/common.go
+++ b/common/common.go
@@ -138,16 +138,12 @@ func NewHTTPClientWithTimeout(t time.Duration) *http.Client {
 // SliceDifference returns the elements that are in slice1 or slice2 but not in both
 func SliceDifference[T comparable](slice1, slice2 []T) []T {
        diff := make([]T, 0, len(slice1)+len(slice2))
-       target := 0
        for x := range slice1 {
                if !slices.Contains(slice2, slice1[x]) {
                        diff = append(diff, slice1[x])
                        continue
                }
-               slice1[target] = slice1[x]
-               target++
        }
-       slice1 = slice1[:target]
        for x := range slice2 {
                if !slices.Contains(slice1, slice2[x]) {
                        diff = append(diff, slice2[x])
```

**Review by shazbert — CHANGES_REQUESTED** [review]
soz

**Review by shazbert — APPROVED** [review]
Thanks tACK.

**Review by gloriousCode — APPROVED** [review]
tACK! 
# 🌞

**Pull request merged**